KeenClient/KIOEventStore.m

```diff
@@ -685,6 +689,12 @@ - (void)closeDB {
 }
 
 - (id)convertNSDateToISO8601:(id)date {
+    __block NSString *iso8601 = nil;
```

Comment:
There shouldn't be any instance where we skip date conversion even if the database is closed.